akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -247,63 +271,107 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
           if (changedKeys.nonEmpty) {
             if (ctx.log.isDebugEnabled) {
               ctx.log.debug(
-                "Change from replicator: [{}], changes: [{}], tombstones [{}]",
+                "ClusterReceptionist [{}] - Change from replicator: [{}], changes: [{}], tombstones [{}]",
+                cluster.selfAddress,
                 newState.entries.entries,
                 changedKeys.map(key ⇒
                   key.asServiceKey.id -> newState.entriesFor(key).mkString("[", ", ", "]")
                 ).mkString(", "),
-                registry.tombstones.mkString(", ")
+                newRegistry.tombstones.mkString(", ")
               )
             }
+
             changedKeys.foreach { changedKey ⇒
-              notifySubscribersFor(changedKey, newState)
+              val serviceKey = changedKey.asServiceKey
+
+              val subscribers = subscriptions.get(changedKey)
+              if (subscribers.nonEmpty) {
+                val listing = ReceptionistMessages.Listing(
+                  serviceKey,
+                  newRegistry.activeActorRefsFor(serviceKey, selfUniqueAddress))
+                subscribers.foreach(_ ! listing)
+              }
```

Comment:
The old logic would filter out tombstoned before notifying to avoid sending out a `Listing` with a removed local ref in the case of concurrent update. While this doesn't, wasn't that a good idea?

Reply:
It still does that in `activeActorRefsFor`. Previous impl only did it for subscription notifications but not for the reply to Register and Find